code/sorting/bubble_sort/bubble_sort.cpp

```diff
@@ -1,82 +1,50 @@
-/* Part of Cosmos by OpenGenus Foundation */
-//Sorting array element (The bubble sort)
-
-#include<iostream>
+/*
+ Part of Cosmos by OpenGenus Foundation
 
-using namespace std;
+ bubble sort synopsis
 
-void bsort(int *,int);               //prototype
-void orderPointer(int *,int *);      //prototype
+template<typename _Bidirectional_Iter, typename _Compare>
+void
+bubbleSort(_Bidirectional_Iter begin, _Bidirectional_Iter end, _Compare compare);
 
-//call by pointer
-void orderPointer(int * a,int * b)
-{
-    if(*a>*b)
-    {
-        int temp=*a;
-        *a=*b;
-        *b=temp;
-    }
+template<typename _Bidirectional_Iter>
+void
+bubbleSort(_Bidirectional_Iter begin, _Bidirectional_Iter end);
+ */
 
-}
+#include <functional>
 
-void bsort(int* ptr,int s)
+template<typename _Bidirectional_Iter, typename _Compare>
+void
+bubbleSort(_Bidirectional_Iter begin, _Bidirectional_Iter end, _Compare compare)
 {
-    for(int i=0; i<s-1 ;++i)
+    if (begin != end)
     {
-        for(int j=i+1; j<s ;++j)
-            orderPointer(ptr+i,ptr+j);
-
-
+        auto frontOfSorted = end;
+
+        for (--frontOfSorted; frontOfSorted != begin; --frontOfSorted)
+        {
+            bool swapped{};
+            for (auto j = begin; j != frontOfSorted; ++j)
+            {
+                auto nextOfJ = j;
+                if (compare(*++nextOfJ, *j))
+                {
+                    std::iter_swap(nextOfJ, j);
+                    swapped = true;
+                }
+            }
+            if (swapped == false)
```

Comment:
if (!swapped)

Reply:
In my opinion the old version is more clear than `!swapped`, since the `!` is not easy to see.

Reply:
I just went by the idea that while(!swapped) could  be read as: while not swapped (makes more sense)